### DISCUSSION TITLE
No anonymous write access

### DISCUSSION THREAD

**Discussion by faha1999** [community/community#23115] [New to GitHub]
<p>From yesterday afternoon every time when I try to commit</p>
<blockquote>
<p>git push origin main:main<br>
remote: No anonymous write access.<br>
fatal: Authentication failed for “repo name”</p>
</blockquote>
<p>why this is happening?<br>
I’m committing to my own repositories. My git account is logged in and everything is okay. But why this is happening?<br>
Are there any solutions for it?</p>

**Comment by Aj0SK** [comment]
<p>I have the same issue for last few hours.</p>

**Comment by krikke01** [comment]
<p>Having the same problem too.</p>

**Comment by Tim-Abwao** [comment]
<p>I’ve had the same issue too since upgrading to version 1.62.2. Searching for a solution led me here.<br>
So far reinstalling and resetting configurations don’t seem to work.</p>
<p>But manual push from CLI using a personal access token works.</p>

**Comment by krikke01** [comment]
<p>Indeed, that works for me as well. Thanks for the tip.</p>

**Comment by Tim-Abwao** [comment]
<p>You’re welcome, <a class="mention" href="/krikke01">@krikke01</a></p>
<p>Also, downgrading to version 1.62.1 seems to work.</p>

**Comment by Andrei9383** [comment]
<p>Having the exact same issue, havent changed anything and it was working before, i’ll try downgrading to 1.62.1 and hope it works</p>
<p>EDIT: Can confirm 1.62.1 works, and also on 1.62.2 if you connect to a WSL session it works</p>

**Comment by ttnsgn** [comment]
<p>I got this message, so just follow the instructions</p>
<blockquote>
<p>remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.<br>
remote: Please see <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" class="inline-onebox" rel="noopener nofollow ugc">Token authentication requirements for Git operations | The GitHub Blog</a> for more information.</p>
</blockquote>

**Comment by KamrulSh** [comment]
<p>I have also faced this problem. I think, It is a vs code issue. I have tried using pycharm. It works fine.</p>

**Comment by dauddi** [comment]
<p>Having the same problem but it seems to be a vs code issue.</p>

**Comment by hardyrafael17** [comment]
<p>I have the same problem, but not only on vscode, all of a sudden I’m unable to push through the terminal, only on repositories which I have clone with through ssh</p>

**Comment by faha1999** [comment, marked as answer]
<p><strong>I just solved this problem.</strong></p>
<p><strong>1st Step</strong>  enable two-factor authentication</p>
<ul>
<li>Go to GitHub Account settings</li>
<li>Account Security</li>
<li>Enable Two-factor authentication</li>
</ul>
<p><strong>2nd step</strong> Personal access tokens</p>
<ul>
<li>Go to GitHub Developer settings</li>
<li>Personal access tokens then generate token</li>
</ul>
<blockquote>
<p>Important notes</p>
</blockquote>
<ol>
<li>Every time when we need to use a <em><strong>GitHub password</strong></em>, we must need to use <strong>Personal access tokens</strong> Instead of the git password.</li>
<li>
<strong>Personal access tokens need to save it in a text file in your machine</strong> because only one time we see the access token.</li>
</ol>
<p>I’m using Linux, so for the <strong>git push</strong>, I’m running this command <strong>sudo git push</strong> Instead of git push.</p>

**Reply by Cedric921** [reply]
it work for me, thanks

**Reply by Umer-309** [reply]
It works for me too, Thanks a lot

**Reply by Lucas3oo** [reply]
this is not working PAT is only for github API

**Reply by naru70o** [reply]
helpful, thanks

**Comment by airtower-luna** [comment]
<p>Two security warnings here: :warning:</p>
<aside class="quote no-group" data-username="faha1999" data-post="12" data-topic="211937">
<div class="title">
<div class="quote-controls"></div>
 faha1999:</div>
<blockquote>
<p>I’m running this command <strong>sudo git push</strong> Instead of git push.</p>
</blockquote>
</aside>
<p>Don’t use <code>sudo</code> for Git. It shouldn’t help unless something’s fundamentally wrong with the ownership/access rights for your repository (and in that case, fix those), but running things with root privileges that don’t need them is always an unnecessary risk.</p>
<aside class="quote no-group" data-username="faha1999" data-post="12" data-topic="211937">
<div class="title">
<div class="quote-controls"></div>
 faha1999:</div>
<blockquote>
<p><strong>Personal access tokens need to save it in a text file in your machine</strong> because only one time we see the access token.</p>
</blockquote>
</aside>
<p>It’s highly recommended to save them in encrypted form, e.g. in a password manager or using GPG. Otherwise anyone who can read the text file can push to your repositories.</p>

**Comment by Emdadul-Tareque** [comment]
<p>Just run this command on your vs code terminal: git config --global credential.helper cache<br>
then run git push …</p>

**Comment by JahanzebNawaz** [comment]
<aside class="quote no-group" data-username="Emdadul-Tareque" data-post="14" data-topic="211937">
<div class="title">
<div class="quote-controls"></div>
 Emdadul-Tareque:</div>
<blockquote>
<p>git config --global credential.helper cache</p>
</blockquote>
</aside>
<p>not working at all.<br>
everything working fine on my terminal but just seems an issue in VS Code.<br>
and a big hurdle is to paste the authentication code each time.</p>

**Comment by sagorhasanmehedi** [comment]
<p>Having the same problem too</p>

**Comment by ghost** [comment]
<p>Facing same issue :frowning:</p>

**Comment by Aabw6088** [comment]
<aside class="quote" data-post="1" data-topic="211937">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/no-anonymous-write-access/211937">No anonymous write access</a> <a class="badge-wrapper  bullet" href="/c/github-help/48"><span class="badge-category-bg" style="background-color: #6a737d;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Need help with your GitHub account or the core features of GitHub? This is the place to look for answers.">GitHub Help</span></a>
  </div>
  <blockquote>
    From yesterday afternoon every time when I try to commit 

git push origin main:main 
remote: No anonymous write access. 
fatal: Authentication failed for “repo name” 

why this is happening? 
I’m committing to my own repositories. My git account is logged in and everything is okay. But why this is happening? 
Are there any solutions for it?
  </blockquote>
</aside>

**Comment by JahanzebNawaz** [comment]
<p><strong>Yahooo! Finally an Issue resolved</strong></p>
<p>So there are a few steps you can try and hopefully, it would be fixed for you too.</p>
<p>Open your Settings In VS CODE.</p>
<ol>
<li>Open settings from the left bottom corner (Settings icon) or press <code>CTRL +,</code>
</li>
<li>type <code>Authentication</code> in the search option, it will show something like this.<br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/f/ef73884d54b15bcd928b7ad3ace1713129ebba14.png" data-download-href="/uploads/short-url/yahDPSahnmy0kLNqc5RsDe24lnK.png?dl=1" title="authentication_github" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/34652154/181074929-0a979493-4f0b-4f78-ae17-065c4ed00552.png" alt="authentication_github" data-base62-sha1="yahDPSahnmy0kLNqc5RsDe24lnK" width="690" height="241" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/f/ef73884d54b15bcd928b7ad3ace1713129ebba14_2_690x241.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/f/ef73884d54b15bcd928b7ad3ace1713129ebba14_2_1035x361.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/e/f/ef73884d54b15bcd928b7ad3ace1713129ebba14.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/f/ef73884d54b15bcd928b7ad3ace1713129ebba14_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">authentication_github</span><span class="informations">1267×443 34.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
<li>You need to uncheck Git: Terminal Authentication. (e.g)</li>
<li>Only Github: Github Authentication should be checked.</li>
</ol>
<ul>
<li>Now you need to close the opened terminal (if any). open your new terminal and try with git commands.</li>
</ul>
<p>on my end, everything is working fine.  hope this works for you too. :slight_smile:</p>

**Reply by vprasadreddy** [reply]
Worked for me. Thank you

**Comment by Okpainmo** [comment]
<p>This process worked just perfectly for me. But no need for “sudo” command when using “git push”. Only git push still does it for me. Thanks so much <a class="mention" href="/faha1999">@faha1999</a>.</p>
